packages/components/src/navigation/NavTab.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { memo } from 'react';
+import classNames from 'classnames';
+import { Draggable } from 'react-beautiful-dnd';
+import { vsClose } from '@deephaven/icons';
+import type { NavTabItem } from './NavTabList';
+import Button from '../Button';
+import ContextActions from '../context-actions/ContextActions';
+import { ResolvableContextAction } from '../context-actions';
+
+interface NavTabProps {
+  tab: NavTabItem;
+  onSelect: (key: string) => void;
+  onClose?: (key: string) => void;
+  isActive: boolean;
+  activeRef: React.RefObject<HTMLDivElement>;
+  index: number;
+  isDraggable: boolean;
+  contextActions?: ResolvableContextAction[];
```

Comment:
Should take `ResolvableContextAction | ResolvableContextAction[]` (like `ContextActions actions` prop does